### DISCUSSION TITLE
Several Hops of Message Passing

### DISCUSSION THREAD

**Discussion by ghost** [pyg-team/pytorch_geometric#4956] [Q&A]
If I want to perform a 7-hop message passing using the `MessagePassing` class in PytorchGeometric, do I need to create 7 instances of my _message-passing network_ or should I pass the updated features 7 times through the same instance?
I am sorry if you don't understand the question but I'll be happy to explain more.

**Comment by rusty1s** [comment]
This depends. Usually, one uses different instances such that each instance can better learn which features to aggregate. It is also possible to share the instance (which will share its weights) across layers, but IMO this should only be used if you are facing serious overfitting.
